test/unit/spread.spec.ts

```diff
@@ -527,3 +527,37 @@ test.each(["LuaTable", "LuaMap"])("can spread %s (#1244)", type => {
     expect(result.some(([k, v]) => k === "foo" && v === "bar")).toBe(true);
     expect(result.some(([k, v]) => k === "fizz" && v === "buzz")).toBe(true);
 });
+
+// https://github.com/TypeScriptToLua/TypeScriptToLua/issues/1384
+test.each(["LuaTable", "LuaMap"])("can spread %s (#1384)", type => {
+    const result: Array<[string, string]> = util.testFunction`
+        const tbl = new ${type}();
+        tbl.set("foo", "bar");
+        tbl.set("fizz", "buzz");
+        return [...tbl];
+    `
+        .withLanguageExtensions()
+        .getLuaExecutionResult();
+
+    // We don't know the order so match like this
+    expect(result).toHaveLength(2);
+    expect(result.some(([k, v]) => k === "foo" && v === "bar")).toBe(true);
+    expect(result.some(([k, v]) => k === "fizz" && v === "buzz")).toBe(true);
```

Comment:
This matcher might help here, doesn't care about the order of the outer-most array but DOES care about the order of `["foo", "bar"]`

```suggestion
    expect(result).toContainEqual(["foo", "bar"]);
    expect(result).toContainEqual(["fizz", "buzz"]);
```

Reply:
Thanks, didn't know about this one. That is much nicer indeed.